scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1065 @@
+        then
+            wsrep_log_error "Could not determine binary to run: $CLONE_BINARY"
+            exit $EINVAL
+        fi
+    fi
+
+    #
+    # Find where libs needed for the binary are
+    #
+    CLONE_LIBS="$WSREP_SST_OPT_PLUGIN_DIR"
+
+    # 1. Try plugins dir
+    if [ -z "$CLONE_LIBS" ] && $MY_PRINT_DEFAULTS "mysqld" | grep -q plugin[_-]dir
+    then
+        CLONE_LIBS=$($MY_PRINT_DEFAULTS mysqld | grep -- "--plugin[_-]dir" | cut -d= -f2)
+        # scale up to the first "mysql" occurence
+        while [ ${#CLONE_LIBS} -gt "1" ]
+        do
+            [ `basename $CLONE_LIBS` = "mysql" ] && break
+            CLONE_LIBS=$(dirname $CLONE_LIBS)
+        done
+        if ls $CLONE_LIBS/private/lib* > /dev/null 2>&1
+        then
+            CLONE_LIBS="$CLONE_LIBS/private"
+        else
+            wsrep_log_info "Could not find private libs in '$CLONE_LIBS' from plugin_dir: $($MY_PRINT_DEFAULTS mysqld | grep -- '--plugin_dir' | cut -d= -f2)"
+            CLONE_LIBS=
+        fi
+    fi
+
+    # 2. Try binary path
+    if [ -z "$CLONE_LIBS" ]
+    then
+        CLONE_LIBS=$(dirname $(dirname $CLONE_BINARY))
+        if ls $CLONE_LIBS/lib64/mysql/private/lib* > /dev/null 2>&1
+        then
+            CLONE_LIBS="$CLONE_LIBS/lib64/mysql/private/"
+        elif ls $CLONE_LIBS/lib/mysql/private/lib* > /dev/null 2>&1
+        then
+            CLONE_LIBS="$CLONE_LIBS/lib/mysql/private/"
+        else
+            wsrep_log_info "Could not find private libs by binary name: $(dirname $(dirname $CLONE_BINARY))"
+            CLONE_LIBS=
+        fi
+    fi
+
+    # 3. Try this script path
+    if [ -z "$CLONE_LIBS" ]
+    then
+        CLONE_LIBS=$(dirname $(dirname $0))
+        if ls $CLONE_LIBS/lib64/mysql/private/lib* > /dev/null 2>&1
+        then
+            CLONE_LIBS="$CLONE_LIBS/lib64/mysql/private"
+        elif ls $CLONE_LIBS/lib/mysql/private/lib* > /dev/null 2>&1
+        then
+            CLONE_LIBS="$CLONE_LIBS/lib/mysql/private"
+        else
+            wsrep_log_info "Could not find private libs by script path: $(dirname $(dirname $0))"
+            CLONE_LIBS=
+        fi
+    fi
+
+    if [ -d "$CLONE_LIBS" ]
+    then
+        CLONE_ENV="LD_LIBRARY_PATH=$CLONE_LIBS:${LD_LIBRARY_PATH:-}"
+    else
+        wsrep_log_info "Could not determine private library path for mysqld: $CLONE_LIBS. Leaving LD_LIBRARY_PATH unmodified."
+        CLONE_ENV=""
+    fi
+
+    MODULE="clone_sst"
+    CLONE_SOCK_DIR=`mktemp --tmpdir -d ${MODULE}_XXXXXX`
+
+    CLONE_PID_FILE="$WSREP_SST_OPT_DATA/$MODULE.pid"
+
+    if check_pid_file $CLONE_PID_FILE
+    then
+        CLONE_PID=`cat $CLONE_PID_FILE`
+        wsrep_log_error "cloning daemon already running (PID: $CLONE_PID)"
+        exit 114 # EALREADY
+    fi
+    rm -rf "$CLONE_PID_FILE"
+
+    # If port was not set up explicitly in wsrep_sst_address, check defaults
+    wsrep_log_debug "-> WSREP_SST_OPT_PORT $WSREP_SST_OPT_PORT"
+
+    if [ -n "${WSREP_SST_OPT_PORT:-}" ]
```

Comment:
Nope. You have two instances on Joiner. The one starting the process (running on Port X and with galera). The other running on port Y and without galera. 
So we have two target ports

Reply:
Well, we never connect to the "original" instance, no? Donor sees only temporary clone instance. So the Donor is interested only with temporary instance and its port.

Reply:
I can pass it, but it is already a configurable param in the my.cnf. 
So it is already something variable (not hardcoded) why should we pass it then

Reply:
The point is that it is configurable for Joiner. It is the Joiner who expects connections on this port. And Joiner should send this info to the Donor. Then Donor knows where to connect.

Reply:
After thinking about it a little bit more, in my opinion we don't need `netcat_port` as well. Donor needs just one port that can be configured via `wsrep_sst_receive_address` (and default to 4444).

Justification:
1. netcat_port is used by Joiner to listen for the SST/IST message from Donor
2. Once Joiner receives this message, this communication channel is not needed anymore
3. So the sequence of events should be:
4. Joiner opens port (4444) to read SST/IST message
5. Joiner receives the message from Donor
6. Joiner closes connection on port (4444)
7. Joiner starts clone instance on port (4444)

The above solution has the following advantages:
1. Compatible with already existing "user interface" (the same way of configuration)
2. No additional ports needed, so no action needed from network administrators to open additional port. If it works with pxb_sst, it works with clone_sst
3. out of the box compatible with MTR suite
4. this also solves the problem with not full message written to the file. Joiner should wait for Donor to close the connection. After the connection is closed and netcat exits, the file has all data.

BTW, have you considered using `socat` instead of `netcat`? We already have good examples of `socat` usage in pxb_sst script (handling timeouts, hangups, etc)